src/overlays/ovl_i6/fox_ve2.c

```diff
@@ -0,0 +1,4132 @@
+                Math_SmoothStepToF(&D_i6_801A7F5C, gLight1R, 1.0f, 4.0f, 0.0f);
+                Math_SmoothStepToF(&D_i6_801A7F64, gLight1G, 1.0f, 3.0f, 0.0f);
+                Math_SmoothStepToF(&D_i6_801A7F6C, gLight1B, 1.0f, 3.0f, 0.0f);
+                Math_SmoothStepToF(&D_i6_801A7F74, gAmbientR, 1.0f, 2.0f, 0.0f);
+                Math_SmoothStepToF(&D_i6_801A7F7C, gAmbientG, 1.0f, 2.0f, 0.0f);
+                Math_SmoothStepToF(&D_i6_801A8430, gAmbientB, 1.0f, 2.0f, 0.0f);
+            }
+            Math_SmoothStepToF(&D_800C9C30, 32.0f, 1.0f, 0.5f, 0.0f);
+            Math_SmoothStepToF(&player->unk_034, 0.0f, 0.2f, 3.0f, 0);
+            boss->fwork[8] = 0.1f;
+            Math_SmoothStepToF(&boss->fwork[7], 200.0f, 1.0f, 10.0f, 0);
+            if (boss->timer_052 == 0) {
+                boss->unk_04C++;
+                if (boss->unk_04C >= Animation_GetFrameCount(&D_C002B08)) {
+                    boss->unk_04C = 0;
+                }
+                frameData = Animation_GetFrameData(&D_C002B08, boss->unk_04C, spD0);
+                Math_SmoothStepToVec3fArray(spD0, D_i6_801A7F80, 1, frameData, boss->fwork[9], 100.0f, 0.0f);
+            } else {
+                if (boss->timer_052 < 16) {
+                    boss->unk_04C++;
+                    if (boss->unk_04C >= Animation_GetFrameCount(&D_C033D98)) {
+                        boss->unk_04C = Animation_GetFrameCount(&D_C033D98) - 1;
+                    }
+                }
+                frameData = Animation_GetFrameData(&D_C033D98, boss->unk_04C, spD0);
+                Math_SmoothStepToVec3fArray(spD0, D_i6_801A7F80, 1, frameData, boss->fwork[9], 100.0f, 0.0f);
+                if (boss->timer_052 == 1) {
+                    boss->unk_04C = 0;
+                }
+            }
+            Math_SmoothStepToF(&boss->fwork[9], 0.2f, 1.0f, 0.01f, 0);
+            if (boss->timer_054 == 0) {
+                boss->actionState = 2;
+                boss->unk_04C = 0;
+                boss->fwork[7] = boss->fwork[9] = 0.0f;
+                boss->vel.z = -20.0f;
+            }
+            break;
+        case 2:
+            boss->fwork[6] = -4000.0f;
+            boss->fwork[8] = 0.05f;
+            Math_SmoothStepToF(&boss->fwork[7], 30.0f, 1.0f, 5.0f, 0);
+            boss->fwork[0] = 0.0f;
+            boss->fwork[2] = 0.05f;
+            boss->fwork[3] = 0.0f;
+            boss->fwork[5] = 0.05f;
+            Math_SmoothStepToF(&boss->fwork[1], 10.0f, 1.0f, 0.5f, 0);
+            Math_SmoothStepToF(&boss->fwork[4], 10.0f, 1.0f, 0.5f, 0);
+            Math_SmoothStepToF(&boss->fwork[9], 0.7f, 1.0f, 0.005f, 0);
+            boss->unk_04C++;
+            if (boss->unk_04C >= Animation_GetFrameCount(&D_C033780)) {
+                func_i6_8018D9C0(boss);
+            }
+            frameData = Animation_GetFrameData(&D_C033780, boss->unk_04C, spD0);
+            Math_SmoothStepToVec3fArray(spD0, D_i6_801A7F80, 1, frameData, boss->fwork[9], 100.0f, 0.0f);
```

Comment:
The return of Animation_GetFrameData is actually the limb count of the animation.